conf/config-schema.json

```diff
@@ -0,0 +1,16 @@
+{
+    "type": "object",
+    "properties": {
+        "root": { "type": "boolean" },
+        "globals": { "type": ["object", "array"] },
+        "global": { "type": ["object", "array"] },
```

Comment:
I asked about this in the first comment. Even if it is not documented, it's used in https://github.com/eslint/eslint/blob/c4ffb49d5a5544e1de0f367e594b989a10ea9c44/lib/eslint.js#L453 as a fallback for globals.

It's also used in a dozen or so tests: https://github.com/eslint/eslint/blob/c4ffb49d5a5544e1de0f367e594b989a10ea9c44/tests/lib/rules/no-global-assign.js#L56
https://github.com/eslint/eslint/blob/c4ffb49d5a5544e1de0f367e594b989a10ea9c44/tests/lib/rules/no-undef.js#L26
to list a few.

Re: the array vs object, here is a test using it: https://github.com/eslint/eslint/blob/c4ffb49d5a5544e1de0f367e594b989a10ea9c44/tests/lib/rules/no-param-reassign.js#L26

I'm fine if we decide to remove/deprecate those, but that should be a TSC decision.